bigquery/gcloud/aio/bigquery/job.py

```diff
@@ -103,7 +103,9 @@ async def insert(self, job: Dict[str, Any],
         project = await self.project()
         url = f'{API_ROOT}/projects/{project}/jobs'
 
-        return await self._post_json(url, job, session, timeout)
+        response = await self._post_json(url, job, session, timeout)
+        self.job_id = response['jobReference'].get('jobId')
```

Comment:
Since you can create a `Job` with a given `job_id`, this line will pave over that value with `None` if this is a dry run. Although generally you should use this `Job` object for a single job, I don't see any protection against re-use.

To align with other behaviour in this class (ex. `insert_via_query`), I think we should avoid setting `self.job_id` at all if the `jobId` field is not present as in the case of `dryRun` like you had mentioned.

Reply:
That's true, I was assuming the expected usage pattern was single use. I'm fine with setting only if it exists.